### PR TITLE
fix(web): position caret correctly after inserting skill mention

### LINKED ISSUE #2638: Caret position is incorrect after inserting a skill reference
## Summary

After referencing a skill, the text cursor/caret appears in the wrong position in the input area.

## Current behavior

When a skill is inserted as a reference/mention in the conversation input, the caret does not appear where the user would expect.

This can make the input feel misaligned or confusing when continuing to type after the inserted skill reference.

## Expected behavior

After inserting a skill reference, the caret should appear in the correct logical typing position.

This means:

- the cursor should align with the visible text flow
- typing should continue from the expected insertion point
- the reference chip/text should not cause caret placement glitches

## Steps to reproduce

1. Open a conversation input that supports skill references.
2. Insert or reference a skill in the input.
3. Look at the caret position after the reference is added.
4. Notice that the cursor appears in the wrong place relative to the visible content.

## Notes

- The screenshot highlights the caret appearing at an incorrect position near the referenced skill chips/text.
- This looks like an input/editor caret positioning issue related to inline references.
- The problem may involve chip rendering, inline element measurement, or selection range restoration after insertion.

## Why this matters

If the caret lands in the wrong place, typing becomes confusing and error-prone. Fixing cursor placement would make referenced-skill workflows feel much more stable and polished.

Image

### PR DESCRIPTION
**Pull request by Diyoncrz18**
Fixes #2638

## Why

When a skill reference is picked from the `@` mention popover in the chat composer, the caret does not land at the visible end of the inserted chip — it can sit inside/over the chip — so the next keystrokes start from the wrong place (reported in #2638). The fix makes the selection placement explicit: after the atomic `MentionNode` is inserted, `selectAfterMention(node)` moves the Lexical selection to just outside the chip (parent paragraph, index + 1) *before* the trailing space is inserted, so typing continues at the logical position after the chip + space.

Scope note (review follow-up): per review, the unrelated `e2e/ui/amr-onboarding.test.ts` timing stabilization that rode along earlier has been removed from this PR; it will be proposed separately if still needed. The branch is now a single commit on top of current `main`, touching only the two files shown in the diff.

## What users will see

- Picking a skill from the `@` popover drops the caret immediately after the chip and its trailing space.
- Typing right after picking a mention continues at the correct spot (e.g. `@Deck Builder please`) instead of landing inside or before the chip.
- Nothing else changes: Backspace/Delete still remove the whole chip, and the staged skill list still prunes when a mention is deleted.

## Surface area

- [x] **None** — caret/selection fix inside an existing composer component plus focused regression tests; no new pages, dialogs, menus, settings, endpoints, contracts, CLI flags, or i18n keys.

## Screenshots

N/A — no new UI surface; the change is caret placement in the existing composer. The behavioral acceptance is the manual QA checklist below (requested before merge).

## Bug fix verification

- Test path: `apps/web/tests/components/composer/LexicalComposerInput.test.tsx` — the `#2638` suite (5 tests): caret lands in the trailing-space node after the pill; immediate typing; non-empty composer; multiple mentions; mention deletion prunes `presentEntities`.
- Red/green honesty note: in jsdom, Lexical's `insertNodes` already leaves the model selection after the atomic node, so the automated suite passes with or without the one-line `selectAfterMention(node)` call — that line normalizes real-browser DOM-selection placement, which jsdom does not reconcile and therefore cannot falsify. The suite still genuinely pins the required behavior: with the trailing-space insertion removed, 6 tests in this file fail (the 5 `#2638` tests plus the pre-existing pill test). Visual acceptance for the reported symptom is the manual QA checklist below.
- Wrapped lines: jsdom cannot measure caret geometry on wrapped lines. The logical selection position that wrapped-line correctness depends on is covered by the automated tests; the visual case is in the manual QA checklist.

## Manual QA (requested before merge)

- [ ] Insert a single skill reference; caret is immediately after the chip.
- [ ] Type text immediately; it appears after chip + space.
- [ ] Insert a mention into a composer that already has text.
- [ ] Insert multiple references.
- [ ] Verify on a wrapped line.
- [ ] Delete a skill mention; confirm no stale skill is staged/sent.

## Validation

- `pnpm exec vitest run -c vitest.config.ts --maxWorkers=2 tests/components/composer/LexicalComposerInput.test.tsx` (in `apps/web`) — PASS (18 passed, 1 intentionally skipped IME-composition guard)
- `pnpm --filter @open-design/web typecheck` — PASS
- `pnpm guard` — PASS

### CODE REVIEW

**Comment by lefarcen** [comment]
Thanks @Diyoncrz18 — nice job spelling out the Lexical selection edge case and adding a focused regression test for it. I'll get the normal routing and follow-up in place from here.

**Comment by PerishCode** [comment]
@Diyoncrz18 I'm holding off on generating review comments for nexu-io/open-design#7515 because this pull request has merge conflicts right now.

Please resolve the conflicts with main and push the updated branch. Once that's done, request or wait for the review to run again and I'll take another look.




<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=codex · An autonomous AI dev team for your GitHub repos.</sub>

**Comment by lefarcen** [comment]
🧪 This PR has changes that need a manual QA pass before merge — please hold off self-merging for now; we'll loop QA in once it's merge-ready (and design/product have signed off, where applicable).

**Review by PerishCode — APPROVED** [review]
@Diyoncrz18 I reviewed the full changed ranges for the composer fix. The new call moves the Lexical selection across the atomic mention boundary before inserting the trailing space, using the component’s existing selection helper, and the regression coverage exercises caret placement, continued typing, non-empty content, and repeated mentions. The focused local test could not run in this prepared worktree because dependencies are not installed, while the live CI jobs are still pending; the code and test changes themselves are coherent and well scoped. Nice work isolating this subtle editor-selection issue and documenting it with targeted coverage.



<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=codex · An autonomous AI dev team for your GitHub repos.</sub>

File: e2e/ui/amr-onboarding.test.ts [hunk 1]
```diff
@@ -139,6 +139,7 @@ test('[P0] Cloud status loading does not block signed-out Local CLI or BYOK setu
 
   await seedOnboardingConfig(page, config);
   await page.goto('/onboarding', { waitUntil: 'domcontentloaded' });
+  await waitForLoadingToClear(page);
```
Comment (PerishCode) [inline e2e/ui/amr-onboarding.test.ts]: Please remove these AMR onboarding timing changes from this PR and land them in a separate focused change. This PR is scoped to the Lexical skill-mention caret bug in `apps/web`, while this line and the matching addition in the next test alter two unrelated P0 onboarding scenarios in a different subsystem; neither change validates the mention fix. Keeping the flake repair here makes the bugfix harder to review and revert independently and conflicts with the repository's focused-bugfix scope rule. A separate PR can document the onboarding flake, its failure evidence, and the appropriate Playwright validation.


<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=codex · An autonomous AI dev team for your GitHub repos.</sub>

**Review by PerishCode — CHANGES_REQUESTED** [review]
The caret fix and its focused component coverage are coherent, but this focused bugfix also includes an unrelated AMR onboarding test stabilization. Please split that change so this PR remains scoped to issue #2638.



<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=codex · An autonomous AI dev team for your GitHub repos.</sub>

**Comment by lefarcen** [comment]
Thanks — @PerishCode's current blocking review is the right next step here. Please drop the unrelated AMR onboarding timing changes from this PR and keep this one focused on the #2638 caret fix; once that split is done, we can re-check the composer change on its own.
